### PR TITLE
wandb chart config map

### PR DESCRIPTION
**Pull request by adityachoudhari26**

### CODE REVIEW

File: charts/operator/values.yaml [hunk 1]
```diff
@@ -32,3 +32,6 @@ clusterRole:
     - apiGroups: ["*"]
       resources: ["*"]
       verbs: ["*"]
+
+localRelease: false
```
Comment (jsbroks) [inline charts/operator/values.yaml:36]: I don't think we need this variable

File: charts/operator/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{{- if .Values.localRelease }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: wandb-chart-configmap
+data:
+  # this should already be base64 encoded
+  operator.tgz: |
+    {{ .Values.wandbChartBinary }}
```
Comment (jsbroks) [inline charts/operator/templates/configmap.yaml:9]:
```suggestion
  wandb-chart.tgz: {{ .Files.Get (.Value.chartPath) | b64enc }}
```

Maybe we could do something like that?

File: charts/operator/values.yaml [hunk 3]
```diff
@@ -32,3 +32,5 @@ clusterRole:
     - apiGroups: ["*"]
       resources: ["*"]
       verbs: ["*"]
+
+chartPath: ""
```
Comment (jsbroks) [inline charts/operator/values.yaml:36]:
```suggestion
charts:
   MY_CHART: './path/tar.gz'
```

File: charts/operator/templates/configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{{- if ne .Values.chartPath "" }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: wandb-chart-configmap
```
Comment (jsbroks) [inline charts/operator/templates/configmap.yaml:5]: You need to prefix stuff with chart name or you can get conficts

File: charts/operator/templates/deployment.yaml [hunk 5]
```diff
@@ -42,6 +42,8 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: spec.serviceAccountName
+        - name: LOCAL_RELEASE
+          value: {{ .Values.localRelease | quote }}
```
Comment (jsbroks) [inline charts/operator/templates/deployment.yaml:46]:
```suggestion
        - name: IS_AIRGAPPED
          value: {{ .Values.airgapped | quote }}
```

File: charts/operator/templates/configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{{- if .Values.airgapped }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "name" . }}-chart-configmap
+data:
+  {{- range .Values.charts }}
+    {{ .name }}.b64: {{ $.Files.Get .path | b64enc }}
```
Comment (adityachoudhari26) [inline charts/operator/templates/configmap.yaml:8]: we are explicitly annotating with `.b64` - in operator code, we check if the file has this extension and handle accordingly

File: charts/operator/templates/deployment.yaml [hunk 7]
```diff
@@ -42,6 +42,8 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: spec.serviceAccountName
+        - name: AIRGAPPED
```
Comment (jsbroks) [inline charts/operator/templates/deployment.yaml:45]:
```suggestion
        - name: `IS_AIRGAPPED`
```

File: charts/operator/templates/deployment.yaml [hunk 8]
```diff
@@ -55,6 +57,11 @@ spec:
           initialDelaySeconds: 5
           periodSeconds: 10
         resources: {{- toYaml .Values.resources | nindent 10 }}
+        {{- if .Values.airgapped }}
+        volumeMounts:
+        - name: {{ include "name" . }}-chart-configmap
```
Comment (jsbroks) [inline charts/operator/templates/deployment.yaml:62]:
```suggestion
        - name: {{ include "name" . }}-charts
```

File: charts/operator/templates/deployment.yaml [hunk 9]
```diff
@@ -66,5 +73,11 @@ spec:
         - containerPort: 8443
           name: https
           protocol: TCP
+      {{- if .Values.airgapped }}
+      volumes:
+      - name: {{ include "name" . }}-chart-configmap
+        configMap:
+          name: {{ include "name" . }}-chart-configmap
```
Comment (jsbroks) [inline charts/operator/templates/deployment.yaml:80]:
```suggestion
      - name: {{ include "name" . }}-charts
        configMap:
          name: {{ include "name" . }}-charts
```

File: charts/operator/values.yaml [hunk 10]
```diff
@@ -32,3 +32,6 @@ clusterRole:
     - apiGroups: ["*"]
       resources: ["*"]
       verbs: ["*"]
+
+airgapped: false
+charts: []
```
Comment (jsbroks) [inline charts/operator/values.yaml:37]: List can we do key string? and iterate object?
```suggestion
charts: {}
```

**Review by jsbroks — CHANGES_REQUESTED** [review]

File: charts/operator/templates/configmap.yaml [hunk 11]
```diff
@@ -0,0 +1,10 @@
+{{- if .Values.airgapped }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "name" . }}-chart-configmap
```
Comment (jsbroks) [inline charts/operator/templates/configmap.yaml:5]:
```suggestion
  name: {{ include "name" . }}-charts
```

File: charts/operator/templates/deployment.yaml [hunk 12]
```diff
@@ -42,6 +42,8 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: spec.serviceAccountName
+        - name: IS_AIRGAPPED
+          value: {{ .Values.airgapped | quote }}
```
Comment (jsbroks) [inline charts/operator/templates/deployment.yaml:46]: bra -.-

**Review by jsbroks — APPROVED** [review]

**Pull request merged**